src/cript/nodes/supporting_nodes/file.py

```diff
@@ -152,25 +153,50 @@ def __init__(self, name: str, source: str, type: str, extension: str = "", data_
 
         super().__init__(name=name, notes=notes, **kwargs)
 
-        # always giving the function the required str regardless if the input `Path` or `str`
-        if _is_local_file(file_source=str(source)):
-            # upload file source if local file
-            source = _upload_file_and_get_object_name(source=source)
-
         # TODO check if vocabulary is valid or not
         # is_vocab_valid("file type", type)
 
         # setting every attribute except for source, which will be handled via setter
         self._json_attrs = replace(
             self._json_attrs,
             type=type,
-            source=source,
+            # always giving the function the required str regardless if the input `Path` or `str`
+            source=str(source),
             extension=extension,
             data_dictionary=data_dictionary,
         )
 
         self.validate()
 
+    def ensure_uploaded(self, api=None):
```

Comment:
does this need to be a public method or can it be private?

Reply:
can we remove passing of the API and just have it use the global API?

Reply:
Yes, if a user is aware of this functionality, I wouldn't want to limit them from optimization.
So, making it public allows the user to take control, when it is beneficial.
I wouldn't want to limit able users.
(The doc string states, that it will be called automatically.)

Reply:
I would like to leave the API in there. Exactly for the public usage.
It appears more transparent to the user if they can choose the API.

And it gives them more options. For the auto upload and the uninformed user, it exactly looks as before.

Also, it just makes sense, that if you call `api.save(project)` that the `api` object is used to perform the upload.
So if we don't rely on the global cache option there, we shouldn't rely on it for the upload either.